### PR TITLE
Add snapshot tests for bitmask flags

### PR DESCRIPTION
**Pull request by Craig-J**
## Summary

_Summary of the purpose of this pull request._

Add tests for Bitmask Flags

Closes TF-637

### Tests Added

_Summary of the tests added as part of this change, or reason why no tests are required._

Snapshot tests for bitmask flags in tooling types. These tests ensure that behaviour remains consistent.

## Contributor Tasks

_These tasks are for the pull request creator to tick off._

- [x] PO review (optional depending on work type)
- [x] XDR review (optional depending on work type)
- [x] QA review (or another developer if no QA is available)
- [x] Ensure documentation requirements are met e.g., public API is commented
- [x] Relevant changelogs have been updated with user-visible changes
    - [x] [TouchFree Windows](/ultraleap/touchfree/blob/-/CHANGELOG-windows.md)
    - [x] [TouchFree BrightSign](/ultraleap/touchfree/blob/-/CHANGELOG-brightsign.md)
    - [x] [TouchFree Web Tooling](/ultraleap/touchfree/blob/-/TF_Tooling_Web/CHANGELOG.md)
- [x] Consider any licensing/other legal implications e.g., notices required

If there is an associated JIRA issue:
- [x] Include a link to the JIRA issue in the summary above
- [x] Make sure the fix version on the issue is set correctly

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

- [ ] Developer testing
- [ ] Code reviewed
- [ ] Non-code assets reviewed
- [ ] Documentation reviewed - includes checking documentation requirements are met and not missing e.g., public API is commented

### CODE REVIEW

File: TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    // Nothing set
+    BitmaskFlags.NONE
+];
+
+describe('BitmaskFlag', () => {
+    it('should combine the same as before', () => {
+        expect(FlagUtilities.GetInteractionFlags).toVerifyAllCombinations(
+            interactionTypes,
+            handTypes,
+            handChiralities,
```
Comment (cartmouse) [inline TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:72]: 🙈 Add "Chiralities" to the skipWords in the eslintrc

File: TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    // Multiple of same enum set
+    allHandChiralityBitmask,
+    allHandTypeBitmask,
+    allInputTypeBitmask,
+    allInteractionTypeBitmask,
+    // Nothing set
+    BitmaskFlags.NONE
+];
+
+describe('BitmaskFlag', () => {
```
Comment (cartmouse) [inline TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:67]: 💬 The tests spit out a _lot_ of errors which is the correct behaviour but clogs up the logs. Consider adding the following to supress them for this test suite only:
```
    beforeAll(() => {
        jest.spyOn(console, 'error').mockImplementation(() => {});
    });
```

File: TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import { BitmaskFlags, FlagUtilities, HandChirality, HandType, InputType, InteractionType } from '../TouchFreeToolingTypes';
```
Comment (cartmouse) [inline TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts]: 🙈 This whole file needs a linting run!

File: TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            interactionTypes,
+            handTypes,
+            handChiralities,
+            inputTypes
+        );
+
+    });
+
+    it('should deserialize hand chirality the same as before', () => {
+        expect(FlagUtilities.GetChiralityFromFlags).toVerifyAllCombinations(bitmaskFlagParams);
```
Comment (cartmouse) [inline TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:78]: 👍 I think this "snapshot" style of testing is perfect in this situation. Will be much easier to maintain as opposed to manually written tests for every combination

File: TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts (hunk 2 above)
Comment (Craig-J) [reply TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:67]: I agree the error spam is bad. I think these errors are not good design - they don't represent erroneous states but are meant to be informative. Should the errors themselves be changed?
Ideally the flag utilities would return `EnumType | ErrorType` or something similar, throw or don't log at all (potentially with an extra defaulted param for the value to return if it cant deserialize).

Comment (cartmouse) [reply TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:67]: I think you're right that the behaviour should probably change. Might be worth creating a new ticket to clean this up with either Error returns or throws as I think it's probably out of scope for this work!

Comment (Craig-J) [reply TF_Tooling_Web/src/tests/TouchFreeToolingTypes.test.ts:67]: Added it to TF-1031

**Review by cartmouse — APPROVED** [review]

**Pull request merged**